components/homme/src/theta-l/share/eos.F90

```diff
@@ -210,16 +210,12 @@ subroutine pnh_and_exner_from_eos2(hvcoord,vtheta_dp,dp3d,dphi,pnh,exner,&
 
    if (present(pnh_i_out)) then
       ! boundary values already computed. interpolate interior
+      ! use linear interpolation in hydrostatic pressure coordinate
+      ! if pnh=pi, then pnh_i will recover pi_i
```

Comment:
I don't think this statement is true, since linear interpolation is not the inverse of midpoint averaging.

Reply:
it's true because dp3d=delta pi.